common/lib/xblock/xblock_discussion/xblock_discussion.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+"""
+Discussion XBlock
+"""
+import json
+import logging
+
+from xblockutils.resources import ResourceLoader
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+from xblock.core import XBlock
+from xblock.fields import Scope, String, UNIQUE_ID
+from xblock.fragment import Fragment
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)  # pylint: disable=invalid-name
+
+
+def _(text):
+    """A noop underscore function that marks strings for extraction."""
+    return text
+
+
+@XBlock.needs('user')
+@XBlock.needs('i18n')
+class DiscussionXBlock(XBlock, StudioEditableXBlockMixin):
+    """Provides a discussion forum that is inline with other content in the courseware."""
+
+    discussion_id = String(scope=Scope.settings, default=UNIQUE_ID, force_export=True)
+    display_name = String(
```

Comment:
It is a pity that this isn't inherited from a base class, but there doesn't seem to be a mixin for this as yet.